rfcs/2021-07-19-8216-multiple-pipelines.md

```diff
@@ -0,0 +1,258 @@
+- Pre/post processing of data - The ability to prepare and normalize data.
+- Connecting pipelines together - The ability to take input from another pipeline.
+- Component reuse - The ability to define boilerplate for reuse across many different pipelines. This will likely align with Datadog’s “pipeline catalogue”.
+- Pipeline quotas - The ability to limit how much data a pipeline can send to a sink.
+- Visibility of the components - The ability to give access to any components within the system.
+
+## Pain
+
+- If an admin wants to split responsibilities between `ops` and `devs` by only allowing the `ops` to deal with `sources/sinks` and the `devs`, it's currently not possible.
+
+## User Experience
+
+These changes are providing a way for vector users to split their configuration in a way that improves the collaboration between ops and devs.
+This split will take be made by creating a pipeline configuration file, in which the devs will be able to configure their components.
+The ops will now configure their `sources` and `sinks`, and expose them to be used by the devs.
+The ops will be able to only given access to some of the common components when configuring the pipeline.
+The devs will have the ability to consume the provided components without being able to change the common configuration.
+The users will be able to monitor their pipelines through the `internal_metrics` source.
+
+## Motivation
+
+- Helps Vector to grow organically within an organization by allowing teams to adopt Vector at their own pace without heavy administrator involvement.
+- Reduces the management overhead of devops/SREs by enabling teams to manage their own pipelines (spread the management load).
+
+## Implementation
+
+### How and where the pipelines are stored
+
+To avoid backward incompatibility, pipelines will be loaded from a `pipelines` sub-directory relative to the Vector configuration directory (e.g., `/etc/vector/pipelines`). Therefore, if a user changes the location of the Vector configuration directory they will also change the `pipelines` directory path. They are coupled.
+The `pipelines` directory will contain all pipelines represented as individual files. For simplicity, and to ensure users do not over complicate pipeline management, sub-directories/nesting are not allowed. This is inspired by Terraform's single-level directory nesting, which has been a net positive for simple management of large Terraform projects.
+Each pipeline file is a processing subset of a larger Vector configuration file. Therefore, it follows the same syntax as Vector's configuration (`toml`, `yaml`, and `json`).
+
+### Configuration
+
+Each time Vector will read its configuration file (on boot and when it live reloads), it will first read the pipeline configuration files and associate an `id` attribute corresponding to the pipeline's filename.
+
+A pipeline can be referenced in a configuration file as follows
```

Comment:
To summarize the comments below. For this first version, I would remove any sort of pipeline declaration in the root file. We can cover that when we follow up with access control and restrictions.